### PR TITLE
UX-1935/Created CAPI Control Plane Details Screen

### PR DESCRIPTION
**Pull request by knnguy**
**NOTE: The edit modals have not been completed yet. That will be in a separate PR**

### AWS Control Plane
Screen Shot 2022-08-18 at 5 28 33 PM

Screen Shot 2022-08-18 at 5 28 41 PM

### EKS Control Plane 
Screen Shot 2022-08-18 at 5 27 45 PM
Screen Shot 2022-08-18 at 5 27 54 PM

### Added error handling to the YAML editor
Screen Shot 2022-08-18 at 5 27 18 PM

### CODE REVIEW

File: src/app/core/hooks/useUpdateAction.ts [hunk 1]
```diff
@@ -109,12 +110,13 @@ const useUpdateAction = <
...
-      } catch (error) {
+      } catch (err) {
         updaterPromisesBuffer.current.shift() // Delete the oldest promise in the sequence (FIFO)
-        localDispatch({ type: 'endUpdate', payload: { success: false, error } })
+        localDispatch({ type: 'endUpdate', payload: { success: false, error: err } })
         success = false
+        error = err.err
       }
-      return { success, response }
+      return { success, response, error }
```
Comment (knnguy) [inline src/app/core/hooks/useUpdateAction.ts:119]: @xantorres Is it ok if I pass the error here?

File: src/app/plugins/infrastructure/components/clusters/capi/details/control-plane/ControlPlanePropertiesCard.tsx [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+            {controlPlane?.etcdBackup?.intervalBackupsEnabled && (
+              <TableRows
+                fields={etcdIntervalBackupFields}
+                labelClassName={classes.indentedFields}
+              />
+            )}
+          </>
+        )}
+        {/* Tags */}
+        <Text variant="caption1">Tags</Text>
```
Comment (patil-pratik-87) [inline src/app/plugins/infrastructure/components/clusters/capi/details/control-plane/ControlPlanePropertiesCard.tsx:404]: should we hide the title `Tags` as well for no tags?

**Review by patil-pratik-87 — APPROVED** [review]

File: src/app/core/hooks/useUpdateAction.ts (hunk 1 above)
Comment (patil-pratik-87) [reply src/app/core/hooks/useUpdateAction.ts:119]: I think it should be fine. Still @xantorres  let me know am overlooking something

File: src/app/plugins/infrastructure/components/clusters/capi/details/control-plane/ControlPlanePropertiesCard.tsx (hunk 2 above)
Comment (knnguy) [reply src/app/plugins/infrastructure/components/clusters/capi/details/control-plane/ControlPlanePropertiesCard.tsx:404]: Yep, I'll hide it. Thanks for pointing it out

File: src/app/plugins/infrastructure/components/clusters/capi/details/PropertyCardTable.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import React from 'react'
+import Text from 'core/elements/Text'
+import { makeStyles } from '@material-ui/styles'
+import Theme from 'core/themes/model'
+import { FieldsForCardsProps } from 'core/components/InfoPanel'
+import clsx from 'clsx'
+
+export const TableRow = ({ label, value, labelClassName = '' }) => {
```
[the remark this answers is not in the document]
Comment (knnguy) [reply src/app/plugins/infrastructure/components/clusters/capi/details/PropertyCardTable.tsx:8]: Right. I wasn't aware that the review table was really similar to the one in the details screens. For now, I'll leave it as it. But once I have time, I will use the `ReviewTable` component

**Pull request merged**